dicom-archive/updateMRI_Upload.pl

```diff
@@ -109,81 +123,118 @@
+        $Settings::db[1],
+        $Settings::db[0],
+        $Settings::db[3],
+        $e->errorMessage,
+        $e->errorCode
+    );
+}
+
 ################################################################
 #####check to see if the tarchiveid is already set or not#######
 ######if it's already in the mri_upload table then it will######
 ######generate an error#########################################
 ################################################################
 
 # fetch tarchiveLibraryDir from ConfigSettings in the database
-my $tarchiveLibraryDir = &NeuroDB::DBI::getConfigSetting(
-                            \$dbh,'tarchiveLibraryDir'
-                            );
-# determine tarchive path stored in the database (without tarchiveLibraryDir)
-my $tarchive_path = $tarchive;
-$tarchive_path    =~ s/$tarchiveLibraryDir\/?//g;  
 
-my $where         = " WHERE t.ArchiveLocation =?";
+my $configOB = NeuroDB::objectBroker::ConfigOB->new(db => $db);
 
-if ($globArchiveLocation) {
-    $where = " WHERE t.ArchiveLocation LIKE ?";
-    $tarchive_path = basename($tarchive);
+my $tarchiveLibraryDir;
+try {
+    $tarchiveLibraryDir = $configOB->getTarchiveLibraryDir();
+} catch(NeuroDB::objectBroker::ObjectBrokerException $e) {
+    die sprintf(
+        "Failed to get tarchive library dir from config: %s\n",
+        $e->errorMessage
+    );
+}
+# determine tarchive path stored in the database (without tarchiveLibraryDir)
+my $tarchive_path = $tarchive;
+$tarchive_path    =~ s/$tarchiveLibraryDir\/?//g;
+
+# Check if there is already an mri upload record for the tarchive
+my $resultRef;
+my $mriUploadOB = NeuroDB::objectBroker::MriUploadOB->new(db => $db);
+try{
+    $resultRef = $mriUploadOB->getWithTarchive(
+        GET_COUNT, $tarchive_path, $globArchiveLocation
+    );
+} catch(NeuroDB::objectBroker::ObjectBrokerException $e) {
+    die sprintf("%s\n", $e->errorMessage);
 }
 
-($query = <<QUERY) =~ s/\n/ /gm;
-SELECT 
-  COUNT(*) 
-FROM 
-  mri_upload m 
-JOIN 
-  tarchive t ON (t.TarchiveID=m.TarchiveID) 
-QUERY
-$query .= $where;
-$sth = $dbh->prepare($query);
-$sth->execute($tarchive_path);
-my $count = $sth->fetchrow_array;
-if($count>0) {
-   print "\n\tERROR: the tarchive is already uploaded \n\n"; 
+if($resultRef->[0]->[0] > 0) {
+   print "\n\tERROR: the tarchive is already uploaded \n\n";
    exit 6;
-} 
-
+}
 
 ################################################################
 #####get the tarchiveid from tarchive table#####################
 ################################################################
-($query = <<QUERY) =~ s/\n/ /gm;
-SELECT 
-  t.TarchiveID 
-FROM 
-  tarchive t 
-QUERY
-$query .= $where;
-$sth = $dbh->prepare($query);
-$sth->execute("%".$tarchive_path."%");
-my $tarchiveID = $sth->fetchrow_array;
 
+my $tarchiveOB = NeuroDB::objectBroker::TarchiveOB->new(db => $db);
+try {
+    $resultRef = $tarchiveOB->getByTarchiveLocation(
+        ['TarchiveID'], $tarchive_path, BASENAME_MATCH
```

Comment:
Technically, shouldn't we be using the $globLocation here as well instead of BASENAME_MATCH?

Reply:
You're absolutely right. Good catch!